ha/core/controllers/pcs/node_controller.py

```diff
@@ -63,26 +63,27 @@ def stop(self, nodeid: str, timeout: int = -1) -> dict:
                 status: Succeeded, Failed, InProgress
         """
         timeout = const.NODE_STOP_TIMEOUT if timeout < 0 else timeout
-        node_status = self.nodes_status([nodeid]).get(nodeid)
-        if node_status == NODE_STATUSES.CLUSTER_OFFLINE.value:
-            Log.info(f"For stop {nodeid}, Node already in offline state.")
-            status = f"Node {nodeid} is already in offline state."
-        elif node_status == NODE_STATUSES.POWEROFF.value:
-            raise ClusterManagerError(f"Failed to stop {nodeid}."
-                f"node is in {node_status}.")
-        else:
-            if self.heal_resource(nodeid):
-                time.sleep(const.BASE_WAIT_TIME)
-            try:
-                Log.info(f"Please Wait, trying to stop node: {nodeid}")
-                self._execute.run_cmd(const.PCS_STOP_NODE.replace("<node>", nodeid)
-                        .replace("<seconds>", str(timeout)))
-                Log.info(f"Executed node stop for {nodeid}, Waiting to stop resource")
-                time.sleep(const.BASE_WAIT_TIME)
-                status = f"Stop for {nodeid} is in progress, waiting to stop resource"
-            except Exception as e:
-                raise ClusterManagerError(f"Failed to stop {nodeid}, Error: {e}")
-        return {"status": const.STATUSES.IN_PROGRESS.value, "output": status, "error": ""}
+        if self.is_valid_node_id(node_id=nodeid):
```

Comment:
Instead of putting if condition everywhere, a simple call will work.

self.valid_node_id(node_id)

If the node_id is not valid, an exception will be thrown and further code will not be executed.

Reply:
Done